### ISSUE TITLE
Facebook's new ANLI dataset? + got Distilbert to outperform NLI models

### ISSUE THREAD

**Issue by Robert60CB** [UKPLab/sentence-transformers#317]
Hi @nreimers hope you're well! Just wondering if you've checked out facebook's new adversarial nli dataset: https://github.com/facebookresearch/anli

And if there were plans to create an example for fine tuning models on that dataset?

P.S not sure if it's worth sharing, and how - but happy if you think it would benefit everyone. I experimented with training  Distilbert on the NLI dataset and felt that the learning rate used was a bit high by default. By using a smaller rate of 8e-6 and fine tuning on AllNLI for 4 epochs and 0.33% warm_up steps, I get a 0.791 performance on STS dataset. For this round I experimented with different pooling layers - concatenating (I'm not sure if this is silly mathematically as I'm not a technical person) mean_sqrt_len and max pooling. 

![110113404_3065328106849520_7321562599356783277_n](https://user-images.githubusercontent.com/17196884/88055472-6b168580-cba2-11ea-939e-6b0eff9bdba9.png)

**Comment by nreimers** [comment]
Hi @Robert60CB 
Thanks for the pointer of the ANLI dataset. I will have a look. I haven't checked it yet.

Thanks for pointing out the thing with the too high learning rate. Once you are done with your experiments, it would be great if you could share your parameters for the distilbert-base-nli-mean-tokens setup. I could then update the model on the server so that everyone profits from the improved performance.

Best
Nils Reimers

**Comment by Robert60CB** [comment]
Hi @nreimers 

Do parameters mean the config file itself, or hyperparameters used when fine tuning? Not too sure so I'll share everything!

Please find all the files to the winning model here: https://drive.google.com/file/d/1_l1OTpxufFjvIJEGCViRlQCmXRgyF5Ic/view?usp=sharing

After doing layer freezing for Distilbert, I get an accuracy of 0.793

![image](https://user-images.githubusercontent.com/17196884/88239408-a52d7780-ccc7-11ea-9b45-d678b44f871a.png)

Here is the code for model initialisation (for my particular task I need to deal with long text)
![image](https://user-images.githubusercontent.com/17196884/88238424-f8ea9180-ccc4-11ea-967d-034dc0e032ed.png)

Once again, empirically - concatenating both max and mean_sqrt_len pooling seems to do well. The 'feeling' / desired intention when I did this, was to somehow preserve the invariance of mean pooling as well as get the expressivity of max pooling. I'm not sure if this is silly mathematically.

I freeze 2 layers of Distilbert - the feeling is the first layers capture more of the fundamental nature of language and we might not want to mess around with that when doing fine tuning. Further experiments can be done like freezing only 1 layer. Or freezing up to 5 layers. 

Everything else remains the same as the 'train nli example' except the training hyperparameters:
![image](https://user-images.githubusercontent.com/17196884/88238647-8cbc5d80-ccc5-11ea-94a9-e35414d259d4.png)

![image](https://user-images.githubusercontent.com/17196884/88238647-8cbc5d80-ccc5-11ea-94a9-e35414d259d4.png)

With a learning rate of 8e-6, epsilon or 1e-7, 3 epochs, and 33% of warm up steps. 

I don't know what epsilon is, but followed the pattern in your code - I notice epsilon is an order of magnitude 10X lower than learning rate, and decrease it accordingly. For some reason I have a hunch it would be silly to have an epsilon higher than LR.

Hope that helps!

**P.S** I noticed in your NLI data you managed to take it from json-ld form to really nice g.z files - could you share the script for that by any chance so I can also look to the adversarial NLI dataset?  Sorry but I still need to learn how to deal with json files and that would help a lot
